extensions/sql-migration/src/dialog/restartMigration/restartMigrationDialog.ts

```diff
@@ -0,0 +1,177 @@
+import { WizardController } from '../../wizard/wizardController';
+import { getMigrationModeEnum, getMigrationTargetTypeEnum } from '../../constants/helper';
+import * as constants from '../../constants/strings';
+import { ServiceContextChangeEvent } from '../../dashboard/tabBase';
+import { migrationServiceProvider } from '../../service/provider';
+import { getSourceConnectionProfile } from '../../api/sqlUtils';
+
+export class RestartMigrationDialog {
+
+	constructor(
+		private readonly _context: vscode.ExtensionContext,
+		private readonly _serviceContext: MigrationServiceContext,
+		private readonly _migration: DatabaseMigration,
+		private readonly _serviceContextChangedEvent: vscode.EventEmitter<ServiceContextChangeEvent>) {
+	}
+
+	private async createMigrationStateModel(
+		serviceContext: MigrationServiceContext,
+		migration: DatabaseMigration,
+		serverName: string,
+		migrationService: features.SqlMigrationService,
+		location: azureResource.AzureLocation): Promise<MigrationStateModel> {
+
+		const stateModel = new MigrationStateModel(this._context, migrationService);
+		const sourceDatabaseName = migration.properties.sourceDatabaseName;
+		const savedInfo: SavedInfo = {
+			closedPage: 0,
+
+			// DatabaseSelector
+			databaseAssessment: [sourceDatabaseName],
+
+			// SKURecommendation
+			databaseList: [sourceDatabaseName],
+			databaseInfoList: [],
+			serverAssessment: null,
+			skuRecommendation: null,
+			migrationTargetType: getMigrationTargetTypeEnum(migration)!,
+
+			// TargetSelection
+			azureAccount: serviceContext.azureAccount!,
+			azureTenant: serviceContext.azureAccount!.properties.tenants[0]!,
+			subscription: serviceContext.subscription!,
+			location: location,
+			resourceGroup: {
+				id: getFullResourceGroupFromId(migration.id),
+				name: getResourceGroupFromId(migration.id),
+				subscription: serviceContext.subscription!,
+			},
+			targetServerInstance: await getMigrationTargetInstance(
+				serviceContext.azureAccount!,
+				serviceContext.subscription!,
+				migration),
+
+			// MigrationMode
+			migrationMode: getMigrationModeEnum(migration),
+
+			// DatabaseBackup
+			targetDatabaseNames: [migration.name],
+			networkContainerType: null,
+			networkShares: [],
+			blobs: [],
+
+			// Integration Runtime
+			sqlMigrationService: serviceContext.migrationService,
+		};
+
+		const getStorageAccountResourceGroup = (storageAccountResourceId: string): azureResource.AzureResourceResourceGroup => {
+			return {
+				id: getFullResourceGroupFromId(storageAccountResourceId!),
+				name: getResourceGroupFromId(storageAccountResourceId!),
+				subscription: this._serviceContext.subscription!
+			};
+		};
+		const getStorageAccount = (storageAccountResourceId: string): azureResource.AzureGraphResource => {
+			const storageAccountName = getResourceName(storageAccountResourceId);
+			return {
+				type: 'microsoft.storage/storageaccounts',
```

Comment:
extract as constant? I have a feeling this exact one might already exist somewhere

Reply:
yep: https://github.com/microsoft/azuredatastudio/blob/43f97f4f56a52082612261b1398cbbfe153952ac/extensions/azurecore/src/azurecore.d.ts#L367

Reply:
this is a rename of the dialog, not new.